src/android/main.kt

```diff
@@ -174,8 +184,15 @@ open class MainActivity : WebViewActivity() {
     super.onNewIntent(intent)
     val window = this.window
     val action = intent.action
+    val data = intent.data
     val id = intent.extras?.getCharSequence("id")?.toString()
 
+    if (data != null) {
+      window.bridge.emit("applicationurl", """{
+        "url": "$data"
```

Comment:
is this 100% safe?

Reply:
in fact, no it isn't! I need to add a check that the action is either `"android.intent.action.MAIN"` or `"android.intent.action.VIEW"`. We can include a check that the `data` is prefixed with the register application protocol scheme configured in the `socket.ini`